kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataCheckSumReplayMetricsSuite.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults
+import io.delta.kernel.Table
+
+/**
+ * Suite to test the engine metrics when loading Domain Metadata through checksum files.
+ */
+class DomainMetadataCheckSumReplayMetricsSuite extends ChecksumLogReplayMetricsTestBase {
+
+  override protected def loadSnapshotFieldsCheckMetrics(
+      table: Table,
+      engine: MetricsEngine,
+      expJsonVersionsRead: Seq[Long],
+      expParquetVersionsRead: Seq[Long],
+      expParquetReadSetSizes: Seq[Long],
+      expChecksumReadSet: Seq[Long],
+      version: Long = -1): Unit = {
+
+    engine.resetMetrics()
+
+    version match {
+      case -1 => table.getLatestSnapshot(engine).getDomainMetadata("foo")
+      case ver => table.getSnapshotAsOfVersion(engine, ver).getDomainMetadata("foo")
+    }
+
+    assertMetrics(
+      engine,
+      expJsonVersionsRead,
+      expParquetVersionsRead,
+      expParquetReadSetSizes,
+      expChecksumReadSet = expChecksumReadSet)
+  }
+
+  // Domain metadata loads double the parquet read sets because:
+  // 1. First read happens during loading Protocol & Metadata in snapshot construction.
+  // 2. Second read happens specifically for domain metadata loading.
+  override protected def getExpectedParquetReadSetSizes(sizes: Seq[Long]): Seq[Long] = {
```

Comment:
I'm a bit confused -- If there is a checksum, wouldn't we _not_ read from the parquet at all? we would use the .crc for both P&M and for domain metadata?

Reply:
I think the method is not named correctly. It is captures if there is no CRC and we fall back to reading from delta log, what is the expected read count.

IMO in that case we should read all at once from the log, rather than doing the log replay twice. Given CRC is the default from now on, this case should occur low.

Reply:
I renamed it to getExpectedCheckpointReadVersions. This are for the test cases with crc missing, checkpoint exists => use checkpoint